compiler/rustc_llvm/build.rs

```diff
@@ -197,8 +197,11 @@ fn main() {
         cfg.define("LLVM_RUSTLLVM", None);
     }
 
-    if tracked_env_var_os("LLVM_NDEBUG").is_some() {
+    if tracked_env_var_os("LLVM_ASSERTIONS").is_none() {
         cfg.define("NDEBUG", None);
+    }
+
+    if tracked_env_var_os("LLVM_NO_DEBUGINFO").is_some() {
```

Comment:
Maybe we should use `DEBUG` here, [as set by Cargo](https://doc.rust-lang.org/cargo/reference/environment-variables.html#environment-variables-cargo-sets-for-build-scripts), so it's based on the crate debug level rather than the LLVM build. That will make a difference in distro builds that aren't even building LLVM at all.

Reply:
Actually, `cc::Build` should do that automatically, so I think we don't need any manual setting.

Reply:
Per https://docs.rs/cc/1.1.1/cc/struct.Build.html#method.debug this actually already happens automatically, so I think this code should just be removed.

Reply:
Oops, race conditions :)